internal/commands/vorpal-engine.go

```diff
@@ -0,0 +1,125 @@
+package commands
+
+import (
+	"path/filepath"
+
+	"github.com/checkmarx/ast-cli/internal/commands/util/printer"
+	errorConstants "github.com/checkmarx/ast-cli/internal/constants/errors"
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func runScanVorpalCommand() func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		vorpalLatestVersion, _ := cmd.Flags().GetBool(commonParams.VorpalLatestVersion)
+		sourceFlag, _ := cmd.Flags().GetString(commonParams.SourcesFlag)
+
+		scanResult, err := ExecuteVorpalScan(sourceFlag, vorpalLatestVersion)
+		if err != nil {
+			return err
+		}
+
+		err = printer.Print(cmd.OutOrStdout(), scanResult, printer.FormatJSON)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func ExecuteVorpalScan(sourceFlag string, vorpalUpdateVersion bool) (*ScanResult, error) {
+	if sourceFlag == "" {
+		return nil, errors.Errorf(errorConstants.FileSourceFlagIsRequired)
+	}
+
+	if filepath.Ext(sourceFlag) == "" {
+		return nil, errors.New(errorConstants.FileExtensionIsRequired)
+	}
+
+	if vorpalUpdateVersion {
+		return ReturnSuccessfulResponseMock(), nil
+	}
+	return ReturnFailureResponseMock(), nil
```

Comment:
please add a todo or comment that it's no supposed to be failure but it's for mock (that the next dev will not think it's a real logic :) )